src/System.Data.Common/tests/System/Data/DataSetTest2.cs

```diff
@@ -1669,42 +1671,32 @@ public void Merge_MissingConstraint()
             table1.Columns.Add("col1", typeof(int));
             table2.Columns.Add("col1", typeof(int));
 
-            try
+            Assert.Throws<DataException>(() =>
             {
                 DataSet ds1 = ds.Copy();
                 DataSet ds2 = ds.Copy();
                 ds2.Tables[0].Constraints.Add("uc", ds2.Tables[0].Columns[0], false);
                 ds1.Merge(ds2, true, MissingSchemaAction.Error);
-                Assert.False(true);
-            }
-            catch (DataException e)
-            {
-            }
+            });
 
-            try
+            //TODO: InvalidConstraintException at 1687 ...Add("fk"..., what are testing exactly?
+            //"This constraint cannot be added since ForeignKey doesn't belong to table table1."
```

Comment:
Seems to be throwing a derived exception, are we supposed to check for InvalidConstraintException or just any DataExceptions?